Compilation fails on Alpine Linux

/usr/src/tarantool/src/lib/small/test/arena_mt.c:27:18: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  intptr_t seed = (unsigned int) pthread_self();

--- a/test/arena_mt.c
+++ b/test/arena_mt.c
@@ -24,7 +24,7 @@ run(void *p __attribute__((unused)))
 #ifdef __FreeBSD__
 	unsigned int seed = pthread_getthreadid_np();
 #else
-	unsigned int seed = (unsigned int) pthread_self();
+	unsigned int seed = (intptr_t) pthread_self();
 #endif
 	int iterations = rand_r(&seed) % ITERATIONS;
 	pthread_t **slabs = slab_map(&arena);

